Iniciare con un EDA  por cada conjunto de datos y despues de hacer unos cambios , los juntaremos empiezo con el primero:

1) importar librerias necesarias

2)leer conjunto de datos

3) df.shape el size del conjunto de datos

4)df.info = informacion general del df

5) df.describe = informacion descriptivas del df

6) visualizacion de todas las variables del conjunto de datos y ver su distribucion


In [114]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import LabelEncoder,MinMaxScaler

df = pd.read_csv(r"C:\Users\JOSUE\Downloads\hackathon\2017_accidents_causa_conductor_gu_bcn_.csv")

In [115]:
df.shape

(11091, 20)

In [116]:
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11091 entries, 0 to 11090
Data columns (total 20 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Numero_expedient            11091 non-null  object 
 1   Codi_districte              11091 non-null  int64  
 2   Nom_districte               11091 non-null  object 
 3   Codi_barri                  11091 non-null  int64  
 4   Nom_barri                   11091 non-null  object 
 5   Codi_carrer                 11091 non-null  int64  
 6   Nom_carrer                  11091 non-null  object 
 7   Num_postal                  11089 non-null  object 
 8   Descripcio_dia_setmana      11091 non-null  object 
 9   NK_Any                      11091 non-null  int64  
 10  Mes_any                     11091 non-null  int64  
 11  Nom_mes                     11091 non-null  object 
 12  Dia_mes                     11091 non-null  int64  
 13  Hora_dia                    110

Codi_districte    Codi_barri    Codi_carrer   NK_Any       Mes_any   
count    11091.000000  11091.000000   11091.000000  11091.0  11091.000000  \
mean         4.760256     26.603462  241016.328014   2017.0      6.505816   
std          3.048208     22.200196  193940.152293      0.0      3.435091   
min         -1.000000     -1.000000      -1.000000   2017.0      1.000000   
25%          2.000000      8.000000  110009.000000   2017.0      4.000000   
50%          4.000000     19.000000  191204.000000   2017.0      6.000000   
75%          7.000000     43.000000  312400.000000   2017.0     10.000000   
max         10.000000     73.000000  701864.000000   2017.0     12.000000   

            Dia_mes      Hora_dia  Coordenada_UTM_X_ED50   
count  11091.000000  11091.000000           11091.000000  \
mean      15.802362     13.760346          430116.605946   
std        8.757472      5.300569            2030.298735   
min        1.000000      0.000000          424208.090000   
25%        8.000000     10.000000          428614.300000   
50%       16.000000     14.000000          430155.920000   
75%       23.000000     18.000000          431572.690000   
max       31.000000     23.000000          435119.930000   

       Coordenada_UTM_Y_ED50      Longitud       Latitud  
count           1.109100e+04  11091.000000  11091.000000  
mean            4.583624e+06      2.162875     41.399275  
std             2.346035e+03      0.024180      0.021200  
min             4.575198e+06      2.091855     41.323094  
25%             4.582190e+06      2.144985     41.386288  
50%             4.583336e+06      2.163448     41.396713  
75%             4.584891e+06      2.180188     41.410826  
max             4.591264e+06      2.222572     41.468244

In [117]:
# def plot_variables(df):
#     # Obteniendo listas de columnas por tipo de datos
#     numeric_cols = df.select_dtypes(include=['int', 'float']).columns
#     categorical_cols = df.select_dtypes(include=['object']).columns
#     boolean_cols = df.select_dtypes(include=['bool']).columns

#     # Graficar variables numéricas
#     for col in numeric_cols:
#         plt.figure(figsize=(8, 6))
#         plt.hist(df[col], bins=20)
#         plt.xlabel(col)
#         plt.ylabel('Frecuencia')
#         plt.title(f'Histograma de {col}')
#         plt.show()

#     # Graficar variables categóricas
#     for col in categorical_cols:
#         plt.figure(figsize=(8, 6))
#         df[col].value_counts().plot(kind='bar')
#         plt.xlabel(col)
#         plt.ylabel('Frecuencia')
#         plt.title(f'Gráfico de barras de {col}')
#         plt.show()

#     # Graficar variables booleanas
#     for col in boolean_cols:
#         plt.figure(figsize=(8, 6))
#         df[col].value_counts().plot(kind='bar')
#         plt.xlabel(col)
#         plt.ylabel('Frecuencia')
#         plt.title(f'Gráfico de barras de {col}')
#         plt.show()


# plot_variables(df)

 Revisaremos si hay nulos u duplicados y si es asi lo solucionaremos de una vez

In [118]:
df.isnull().sum()

Numero_expedient              0
Codi_districte                0
Nom_districte                 0
Codi_barri                    0
Nom_barri                     0
Codi_carrer                   0
Nom_carrer                    0
Num_postal                    2
Descripcio_dia_setmana        0
NK_Any                        0
Mes_any                       0
Nom_mes                       0
Dia_mes                       0
Hora_dia                      0
Descripcio_torn               0
Descripcio_causa_conductor    0
Coordenada_UTM_X_ED50         0
Coordenada_UTM_Y_ED50         0
Longitud                      0
Latitud                       0
dtype: int64

In [119]:
# la borramos ya que tenemos muchas informacion DESCRIPTIVA repetitiva y esta es una y no ayudara
df.drop('Num_postal ', axis=1, inplace=True)

In [120]:
df.duplicated().sum()

0

Anomalias lo resolveremos:

Revisaremos la variable desconegut y no hay informacion 93 de los 104 datos son nulos borraremos esas fila,despues de borrar filas reseteamos el index ,tambien encontramos anomalias en los nombre de las variables las pasaremos a formato minuscula

In [121]:
filtro = df['Nom_districte'] == 'Desconegut'
conjunto_filtrado = df[filtro]


In [122]:
conjunto_filtrado = conjunto_filtrado.reset_index(drop=True)

In [123]:
conjunto_filtrado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 104 entries, 0 to 103
Data columns (total 19 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Numero_expedient            104 non-null    object 
 1   Codi_districte              104 non-null    int64  
 2   Nom_districte               104 non-null    object 
 3   Codi_barri                  104 non-null    int64  
 4   Nom_barri                   104 non-null    object 
 5   Codi_carrer                 104 non-null    int64  
 6   Nom_carrer                  104 non-null    object 
 7   Descripcio_dia_setmana      104 non-null    object 
 8   NK_Any                      104 non-null    int64  
 9   Mes_any                     104 non-null    int64  
 10  Nom_mes                     104 non-null    object 
 11  Dia_mes                     104 non-null    int64  
 12  Hora_dia                    104 non-null    int64  
 13  Descripcio_torn             104 non

In [124]:
conjunto_filtrado['Nom_carrer'].value_counts()

Nom_carrer
Desconegut                                            93
Corts Catalanes                                        2
Motors                                                 1
Joan XXIII / Martí i Franquès                          1
Número 6 Zona Franca / A Zona Franca                   1
Aguilar                                                1
Zona Franca                                            1
Aribau                                                 1
Sentís                                                 1
Mare de Déu de Port                                    1
Francesc Ferrer i Guàrdia                              1
Name: count, dtype: int64

In [125]:
# borraremos la filas Desconegut ya que no aportan
df = df[df['Nom_districte'] != 'Desconegut']


In [126]:
# reseteamos el indice df3
df = df.reset_index(drop=True)

In [127]:
# cambiamos a minusculas todos los nombres de las variables para hacer el merge
df.columns = df.columns.str.lower()

In [128]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10987 entries, 0 to 10986
Data columns (total 19 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   numero_expedient            10987 non-null  object 
 1   codi_districte              10987 non-null  int64  
 2   nom_districte               10987 non-null  object 
 3   codi_barri                  10987 non-null  int64  
 4   nom_barri                   10987 non-null  object 
 5   codi_carrer                 10987 non-null  int64  
 6   nom_carrer                  10987 non-null  object 
 7   descripcio_dia_setmana      10987 non-null  object 
 8   nk_any                      10987 non-null  int64  
 9   mes_any                     10987 non-null  int64  
 10  nom_mes                     10987 non-null  object 
 11  dia_mes                     10987 non-null  int64  
 12  hora_dia                    10987 non-null  int64  
 13  descripcio_torn             109

 Ya que sabemos que tenemos 10 distritos y 73 barrios tocaba ver cuantas calles hay pero me quedare con el codigo de calle ya que hara menOs ruidos a que despues tenga que encodificar todas las calles LA BORRARE

In [129]:
with pd.option_context('display.max_rows', None, 'display.max_columns', None):
    print(df['nom_carrer'].value_counts())

nom_carrer
Corts Catalanes                449
Diagonal                       388
Aragó                          278
Litoral (Llobregat)            220
València                       199
Meridiana                      176
Dalt (Besòs)                   165
Mallorca                       164
Litoral (Besòs)                139
Gràcia                         126
Balmes                         117
Còrsega                        108
Dalt (Llobregat)                97
Diputació                       95
Muntaner                        93
Consell de Cent                 92
Provença                        92
Marina                          90
Rosselló                        85
General Mitre                   79
Paral·lel                       75
Lepant                          74
Comte d'Urgell                  73
Sant Joan                       70
Augusta                         67
Sants                           66
Aribau                          66
Catalunya                       66
Sardenya 

In [130]:
# VALORES UNICOS
Valores_unicos = df['nom_carrer'].unique()
cantidad_valores_unicos = len(Valores_unicos)
print(cantidad_valores_unicos)

1584


1584 calles en la ciudad de barcelona en los 10 distritos y 73 barrios, lo que hare sera crear una columna con los nombre de las calles encodificadas utilizando label encoder de sklearn y eso me ayudara a no perder la variable ya que es categorica y no quiero hacerla getdummies porque me crearia 1584 variables binarias y no es muy apropiado ,y despues que haga la encodificacion verificare la imformacion para ver si se a ejecutado correctamente y los valores tienen que ser iguales al contenido original

In [131]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()
df['carrer_encoded'] = encoder.fit_transform(df['nom_carrer'])
df = df.drop(['nom_carrer'], axis=1)  # Elimina la columna original 'Calle'


In [132]:
with pd.option_context('display.max_rows', None, 'display.max_columns', None):
    print(df['carrer_encoded'].value_counts())

carrer_encoded
380     449
417     388
75      278
771     220
1512    199
916     176
402     165
828     164
770     139
635     126
125     117
398     108
403      97
419      95
978      93
366      92
1159     92
863      90
1246     85
612      79
1064     75
762      74
353      73
1311     70
101      67
1356     66
87       66
298      66
1362     66
206      63
1284     59
1077     58
844      57
1071     52
670      51
1169     49
1044     47
516      47
471      46
1537     45
925      45
769      44
289      43
1495     41
379      41
1363     40
1430     40
401      39
510      38
1498     38
753      38
1551     37
853      37
297      37
117      36
1378     34
1575     34
814      33
1008     33
617      33
739      32
1237     32
1157     32
287      31
1290     30
641      30
121      30
1500     30
1009     29
1375     29
352      28
223      28
693      28
669      28
644      27
496      27
1180     26
177      26
519      26
850      25
175      25
489      25
2

In [133]:
# VALORES UNICOS
Valores_unicos = df['carrer_encoded'].unique()
cantidad_valores_unicos = len(Valores_unicos)
print(cantidad_valores_unicos)

1584


In [134]:
df.head(5)

numero_expedient  codi_districte nom_districte  codi_barri   
0  2017S010628                  10    Sant Martí          64  \
1  2017S003663                  10    Sant Martí          64   
2  2017S010680                  10    Sant Martí          64   
3  2017S003193                  10    Sant Martí          64   
4  2017S008857                  10    Sant Martí          64   

                    nom_barri  codi_carrer descripcio_dia_setmana  nk_any   
0  el Camp de l'Arpa del Clot       226400               Diumenge    2017  \
1  el Camp de l'Arpa del Clot       226400                Dilluns    2017   
2  el Camp de l'Arpa del Clot       161407                Dimarts    2017   
3  el Camp de l'Arpa del Clot       194406                 Dijous    2017   
4  el Camp de l'Arpa del Clot       297001                 Dijous    2017   

   mes_any   nom_mes  dia_mes  hora_dia                descripcio_torn   
0       12  Desembre       24         0  Gir indegut o sense precaució  \
1        4     Abril       24        21               Desobeir semàfor   
2       12  Desembre       26        20                 No determinada   
3        4     Abril        6        14                     Desconegut   
4       10   Octubre       26         9               Desobeir semàfor   

  descripcio_causa_conductor  coordenada_utm_x_ed50  coordenada_utm_y_ed50   
0                        Nit              432141.03             4585382.64  \
1                      Tarda              432024.02             4585501.49   
2                      Tarda              431863.93             4585505.67   
3                      Tarda              431629.04             4585416.36   
4                       Matí              431181.73             4585030.22   

   longitud    latitud  carrer_encoded  
0  2.190956  41.409311             758  
1  2.196339  41.413514             758  
2  2.187887  41.414538             670  
3  2.195267  41.416687             844  
4  2.191832  41.409389            1284

In [135]:
# Lista de las columnas a borrar
columnas_a_borrar = ['codi_carrer','nom_barri','nom_districte','latitud', 'longitud', 'coordenada_utm_y_ed50', 'coordenada_utm_x_ed50', 'hora_dia', 'dia_mes', 'nom_mes', 'mes_any', 'nk_any']
# Borra las columnas
df = df.drop(columns=columnas_a_borrar)


In [136]:
df = pd.get_dummies(df, columns=['descripcio_torn','descripcio_dia_setmana', 'descripcio_causa_conductor'], prefix='', prefix_sep='')

In [137]:
numeric_cols = df.select_dtypes(include=['int', 'float']).columns
categorical_cols = df.select_dtypes(include=['object']).columns
boolean_cols = df.select_dtypes(include=['bool']).columns

In [138]:
df[boolean_cols] = df[boolean_cols].astype(int)

In [139]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10987 entries, 0 to 10986
Data columns (total 30 columns):
 #   Column                                   Non-Null Count  Dtype 
---  ------                                   --------------  ----- 
 0   numero_expedient                         10987 non-null  object
 1   codi_districte                           10987 non-null  int64 
 2   codi_barri                               10987 non-null  int64 
 3   carrer_encoded                           10987 non-null  int32 
 4   Altres                                   10987 non-null  int32 
 5   Avançament defectuós/improcedent         10987 non-null  int32 
 6   Canvi de carril sense precaució          10987 non-null  int32 
 7   Desconegut                               10987 non-null  int32 
 8   Desobeir altres senyals                  10987 non-null  int32 
 9   Desobeir semàfor                         10987 non-null  int32 
 10  Envair calçada contrària                 10987 non-null  i

Creare otras columnas para enriquecer el conjunto de datos ya que nuestro objetivo es reducir la dimensionalidad

In [140]:
df['Accidentes_Distrito'] = df.groupby('codi_districte')['numero_expedient'].transform('count')
df['Accidentes_Barrio'] = df.groupby('codi_barri')['numero_expedient'].transform('count')
df['Accidentes_Calle'] = df.groupby('carrer_encoded')['numero_expedient'].transform('count')



In [152]:
df.head(5)

numero_expedient  codi_districte  codi_barri  carrer_encoded  Altres   
0  2017S010628                  10          64             758       0  \
1  2017S003663                  10          64             758       0   
2  2017S010680                  10          64             670       0   
3  2017S003193                  10          64             844       0   
4  2017S008857                  10          64            1284       0   

   Avançament defectuós/improcedent  Canvi de carril sense precaució   
0                                 0                                0  \
1                                 0                                0   
2                                 0                                0   
3                                 0                                0   
4                                 0                                0   

   Desconegut  Desobeir altres senyals  Desobeir semàfor  ...  Dimecres   
0           0                        0                 0  ...         0  \
1           0                        0                 1  ...         0   
2           0                        0                 0  ...         0   
3           1                        0                 0  ...         0   
4           0                        0                 1  ...         0   

   Dissabte  Diumenge  Divendres  Matí  Nit  Tarda  Accidentes_Distrito   
0         0         1          0     0    1      0                 1418  \
1         0         0          0     0    0      1                 1418   
2         0         0          0     0    0      1                 1418   
3         0         0          0     0    0      1                 1418   
4         0         0          0     1    0      0                 1418   

   Accidentes_Barrio  Accidentes_Calle  
0                191                14  
1                191                14  
2                191                51  
3                191                57  
4                191                59  

[5 rows x 33 columns]

In [142]:
# le cambie el nombre por la postrofe que tenia en medio del nombre
df.rename(columns={"Manca d'atenció a la conducció": "Manca de atenció a la conducció"}, inplace=True)


In [153]:
df.shape

(10987, 33)

In [154]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10987 entries, 0 to 10986
Data columns (total 33 columns):
 #   Column                                   Non-Null Count  Dtype 
---  ------                                   --------------  ----- 
 0   numero_expedient                         10987 non-null  object
 1   codi_districte                           10987 non-null  int64 
 2   codi_barri                               10987 non-null  int64 
 3   carrer_encoded                           10987 non-null  int32 
 4   Altres                                   10987 non-null  int32 
 5   Avançament defectuós/improcedent         10987 non-null  int32 
 6   Canvi de carril sense precaució          10987 non-null  int32 
 7   Desconegut                               10987 non-null  int32 
 8   Desobeir altres senyals                  10987 non-null  int32 
 9   Desobeir semàfor                         10987 non-null  int32 
 10  Envair calçada contrària                 10987 non-null  i

In [165]:
group_data1 = df.groupby(['codi_districte', 'codi_barri']).agg({'numero_expedient': 'count','carrer_encoded': 'first','Accidentes_Distrito': 'first', 'Accidentes_Barrio': 'first', 'Accidentes_Calle': 'first', 'Altres': 'sum', 'Avançament defectuós/improcedent': 'sum', 'Canvi de carril sense precaució': 'sum', 'Desconegut': 'sum', 'Desobeir altres senyals': 'sum', 'Desobeir semàfor':  'sum', 'Envair calçada contrària': 'sum', 'Fallada mecànica o avaria': 'sum', 'Gir indegut o sense precaució': 'sum','Manca precaució efectuar marxa enrera': 'sum','Manca precaució incorporació circulació':'sum', 'No determinada':'sum', 'No respectar distàncies': 'sum','Manca de atenció a la conducció': 'sum', 'No respectat pas de vianants': 'sum', 'Dijous':'sum', 'Dilluns': 'sum', 'Dimarts': 'sum', 'Dimecres': 'sum', 'Dissabte': 'sum', 'Diumenge': 'sum', 'Divendres': 'sum', 'Matí': 'sum', 'Nit':'sum', 'Tarda': 'sum'})

In [166]:
 group_data1.shape

(73, 30)

In [167]:
group_data1.head(10)

numero_expedient  carrer_encoded   
codi_districte codi_barri                                     
1              1                        179             447  \
               2                        184             298   
               3                        162             770   
               4                        160            1050   
2              5                        288             380   
               6                        502             287   
               7                       1139             617   
               8                        621            1159   
               9                        389              75   
               10                       264            1542   

                           Accidentes_Distrito  Accidentes_Barrio   
codi_districte codi_barri                                           
1              1                           685                179  \
               2                           685                184   
               3                           685                162   
               4                           685                160   
2              5                          3203                288   
               6                          3203                502   
               7                          3203               1139   
               8                          3203                621   
               9                          3203                389   
               10                         3203                264   

                           Accidentes_Calle  Altres   
codi_districte codi_barri                             
1              1                         11      38  \
               2                         66      39   
               3                        139      26   
               4                         10      21   
2              5                        449      35   
               6                         31      56   
               7                         33     152   
               8                         92      68   
               9                        278      50   
               10                        14      38   

                           Avançament defectuós/improcedent   
codi_districte codi_barri                                     
1              1                                         12  \
               2                                          4   
               3                                          6   
               4                                          3   
2              5                                         11   
               6                                          8   
               7                                         29   
               8                                         22   
               9                                         15   
               10                                        10   

                           Canvi de carril sense precaució  Desconegut   
codi_districte codi_barri                                                
1              1                                         2          18  \
               2                                         8          22   
               3                                        17          17   
               4                                        11          16   
2              5                                        21          19   
               6                                        38          40   
               7                                       106          76   
               8                                        55          30   
               9                                        35          24   
               10                                       26          22   

                           Desobeir altres senyals  ...  Dijous  Dilluns   
codi_districte codi_barri                           ...   

# Creamos un nuevo dataframe con el conjunto de datos agrupado y agregado

In [168]:
df1 = group_data1.reset_index()

In [169]:
df1.shape

(73, 32)

In [170]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73 entries, 0 to 72
Data columns (total 32 columns):
 #   Column                                   Non-Null Count  Dtype
---  ------                                   --------------  -----
 0   codi_districte                           73 non-null     int64
 1   codi_barri                               73 non-null     int64
 2   numero_expedient                         73 non-null     int64
 3   carrer_encoded                           73 non-null     int32
 4   Accidentes_Distrito                      73 non-null     int64
 5   Accidentes_Barrio                        73 non-null     int64
 6   Accidentes_Calle                         73 non-null     int64
 7   Altres                                   73 non-null     int32
 8   Avançament defectuós/improcedent         73 non-null     int32
 9   Canvi de carril sense precaució          73 non-null     int32
 10  Desconegut                               73 non-null     int32
 11  Desobeir

In [172]:
 df1.describe()

codi_districte  codi_barri  numero_expedient  carrer_encoded   
count       73.000000   73.000000         73.000000       73.000000  \
mean         6.246575   37.000000        150.506849      757.013699   
std          2.807633   21.217131        163.646394      435.312451   
min          1.000000    1.000000         13.000000       13.000000   
25%          4.000000   19.000000         54.000000      401.000000   
50%          7.000000   37.000000        106.000000      715.000000   
75%          8.000000   55.000000        191.000000     1068.000000   
max         10.000000   73.000000       1139.000000     1571.000000   

       Accidentes_Distrito  Accidentes_Barrio  Accidentes_Calle      Altres   
count            73.000000          73.000000         73.000000   73.000000  \
mean           1073.671233         150.506849         54.547945   21.561644   
std             708.745450         163.646394         99.120023   21.094356   
min             560.000000          13.000000          1.000000    2.000000   
25%             634.000000          54.000000          3.000000    8.000000   
50%             764.000000         106.000000         11.000000   17.000000   
75%            1206.000000         191.000000         57.000000   31.000000   
max            3203.000000        1139.000000        449.000000  152.000000   

       Avançament defectuós/improcedent  Canvi de carril sense precaució  ...   
count                         73.000000                        73.000000  ...  \
mean                           4.356164                        10.575342  ...   
std                            4.902862                        15.088551  ...   
min                            0.000000                         0.000000  ...   
25%                            1.000000                         2.000000  ...   
50%                            3.000000                         7.000000  ...   
75%                            6.000000                        13.000000  ...   
max                           29.000000                       106.000000  ...   

           Dijous     Dilluns     Dimarts    Dimecres    Dissabte   Diumenge   
count   73.000000   73.000000   73.000000   73.000000   73.000000  73.000000  \
mean    24.520548   22.136986   24.589041   24.123288   16.753425  12.863014   
std     26.635455   24.042968   28.476518   28.934766   18.338470  12.832592   
min      2.000000    0.000000    1.000000    0.000000    0.000000   0.000000   
25%      7.000000    8.000000    7.000000    8.000000    7.000000   5.000000   
50%     18.000000   16.000000   18.000000   18.000000   13.000000  10.000000   
75%     32.000000   28.000000   31.000000   29.000000   21.000000  17.000000   
max    173.000000  164.000000  196.000000  206.000000  128.000000  90.000000   

        Divendres        Matí         Nit       Tarda  
count   73.000000   73.000000   73.000000   73.000000  
mean    25.520548   60.356164   16.931507   73.219178  
std     28.041195   66.045096   18.645322   80.250172  
min      2.000000    3.000000    0.000000    4.000000  
25%      8.000000   22.000000    7.000000   26.000000  
50%     19.000000   46.000000   12.000000   50.000000  
75%     31.000000   75.000000   20.000000   91.000000  
max    182.000000  450.000000  123.000000  566.000000  

[8 rows x 32 columns]

In [174]:
df1.isnull().sum()

codi_districte                             0
codi_barri                                 0
numero_expedient                           0
carrer_encoded                             0
Accidentes_Distrito                        0
Accidentes_Barrio                          0
Accidentes_Calle                           0
Altres                                     0
Avançament defectuós/improcedent           0
Canvi de carril sense precaució            0
Desconegut                                 0
Desobeir altres senyals                    0
Desobeir semàfor                           0
Envair calçada contrària                   0
Fallada mecànica o avaria                  0
Gir indegut o sense precaució              0
Manca precaució efectuar marxa enrera      0
Manca precaució incorporació circulació    0
No determinada                             0
No respectar distàncies                    0
Manca de atenció a la conducció            0
No respectat pas de vianants               0
Dijous    

In [175]:
df1.duplicated().sum()

0

Guardo en un CSV mi conjunto de datos limpio y agregado y listo para el merge y despues la reduccion de dimensiones PCA

In [173]:
df1.to_csv(r'C:\Users\JOSUE\Downloads\hackathon\Accidentes_incidencias_limpio.csv', index=False)